internal/jobs/repo/import.go

```diff
@@ -218,3 +243,111 @@ func difference(existing, new []string) []string {
 
 	return diff
 }
+
+// TODO(ramirocastillo):Move this fn to the helper package
+// fetchGitHubTokenFromDB is a temporary helper function for retrieving the most recently added GITHUB_PAT service credential from the DB.
+// It's "temporary" because the way credentials are managed and retrieved will likely need to be much more robust in the future.
+func fetchGitHubTokenFromDB(ctx context.Context, qry *db.Queries) (string, error) {
+	var credentials string
+	var err error
+	encryptionSecret := os.Getenv("ENCRYPTION_SECRET")
+
+	if credentials, err = qry.FetchGitHubToken(ctx, encryptionSecret); err != nil && !errors.Is(err, pgx.ErrNoRows) {
+		return credentials, fmt.Errorf("could not retrieve GitHub PAT from database: %v", err)
+	}
+
+	if len(credentials) <= 0 {
+		// default to the `GITHUB_TOKEN` env var if nothing in the DB
+		credentials = os.Getenv("GITHUB_TOKEN")
+	}
+
+	return string(credentials), nil
+}
+
+// fetchGitHubReposByOrg fetch all repos from a given organization, if a github token is not
+// provided will search only public repos
+func fetchGitHubReposByOrg(ctx context.Context, client *github.Client, repoOwner string, ghToken string) ([]*githubRepository, error) {
+	var repositories []*githubRepository
+	var repos []*github.Repository
+	var err error
+	var typeOfRepo string
+	var resp *github.Response
+
+	if len(ghToken) <= 0 {
+		typeOfRepo = "public"
+	}
+	opt := &github.RepositoryListByOrgOptions{Type: typeOfRepo}
+
+	for {
+		if repos, resp, err = client.Repositories.ListByOrg(ctx, repoOwner, opt); err != nil {
+			return repositories, err
+		}
+
+		for _, repo := range repos {
+
+			jsonStr, err := json.Marshal(&repo.Topics)
+			if err != nil {
+				return repositories, err
+			}
+
+			repositories = append(repositories, &githubRepository{Name: *repo.Name, Owner: string(*repo.Owner.OrganizationsURL),
+				Topics: string(jsonStr)})
+		}
+		if resp == nil {
+			break
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+		continue
```

Comment:
I don't think we need this continue since we're just at the end of the loop

Reply:
we do .The way the api manages their pagination feature is by `resp.NexPage` , we should always loop to the next iteration is the nextPage from the response !=0 otherwise we break the loop in line 300